### DISCUSSION TITLE
Fitted values in sktime

### DISCUSSION THREAD

**Discussion by saumya-verma-DS** [alan-turing-institute/sktime#6310] [General]
My code is 
from sklearn.neighbors import KNeighborsRegressor
from sktime.forecasting.compose import make_reduction
regressor = KNeighborsRegressor(n_neighbors=5)
forecaster = make_reduction(regressor, strategy="direct", window_length=6)
forecaster.fit(train_targets,fh =[-1])
predictions_train = forecaster.predict(fh = [-1])   

and the error I am getting NotImplementedError: In-sample predictions are not implemented.  But there is no other way to get fitted values in sktime.
My main problem is that I would need fitted values for my models. How can I get those?

**Comment by fkiraly** [comment]
You could try the alternative implementatoin, `DirectReductionForecaster` - it is slower, but can return in-sample forecasts.

**Reply by saumya-verma-DS** [reply]
Thankyou for the help. I will try this out
